### ISSUE TITLE
Feature Request - support tags field in aws_ssm_document

### ISSUE THREAD

**Issue by aluminous** [terraform-providers/terraform-provider-aws#5015]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Please consider adding support for settings tags on `aws_ssm_document`. An example use case is to restrict users to documents based on tags using IAM, as described by Amazon [here](https://docs.aws.amazon.com/systems-manager/latest/userguide/sysman-ssm-docs-tagging.html#sysman-ssm-docs-tagging-access).

### New or Affected Resource(s)

* aws_ssm_document

### Potential Terraform Configuration

```hcl
resource "aws_ssm_document" "default" {
    name = "example"
    document_type = "Command"
    document_format = "YAML"

    content = "${file("my-document.yml")}"

    tags {
        SomeTag = "ExampleValue"
    }
}
```

### References

* #1584 - tagging support recently added to the related `aws_ssm_parameter`

**Comment by bflad** [comment]
Pull request submitted: #5020

**Comment by bflad** [comment]
Support for tagging has been merged into master and will release with version 1.26.0 of the AWS provider, likely middle of next week. 👍

**Comment by bflad** [comment]
This has been released in [version 1.26.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1260-july-04-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
